src/txDigester/ipfsPublisher.ts

```diff
@@ -0,0 +1,124 @@
+import { join } from 'path'
+import { Blob } from 'buffer'
+import { TransactionDigest } from './txDigests'
+import type { Client } from '@web3-storage/w3up-client'
+import { overrideDefaultConfig, config } from '../Config'
+import { mainLogger } from '../Logger'
+import { BlobLike } from '@web3-storage/w3up-client/dist/src/types'
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { W3SClient, Email } = require(join(process.cwd(), '/src/utils/esm/bundle.js'))
+
+const configFile = join(process.cwd(), 'archiver-config.json')
+overrideDefaultConfig(configFile)
+
+const { admin_email, root_did } = config.txDigest.web3Storage
+
+type Web3StorageRootDID = `did:${string}:${string}`
+type Web3StorageAdminEmail = `${string}@${string}`
+
+let client: Client
+let lastUploadTime = 0
+let isPublisherActive = false
+
+// Note: This can be removed in production or in cases where uploads will happen once in 5 or more cycles/minutes
+const REST_PERIOD_BETWEEN_UPLOADS = 1000 * 60 * 5 // 5 minutes
+
+// Upload a file to the specified space DID
+export const uploadDigestToIPFS = async (data: TransactionDigest): Promise<void> => {
+  try {
+    if (isUploadPossible()) {
+      console.log(
+        `❌ Publisher cannot upload to IPFS: ${
+          isPublisherActive ? 'Another Upload in progress.' : 'Rest Period is Active.'
+        }`
+      )
+      return
+    }
+    isPublisherActive = true
+    console.log(`Uploading TX Digest for Cycle Range ${data.cycleStart} to ${data.cycleEnd}`)
+    await client.setCurrentSpace(root_did as Web3StorageRootDID)
+    console.log(`Uploading Data to Root-DID: ${root_did}`)
+
+    const { cycleStart: cs, cycleEnd: ce, txCount: tc, hash: h } = data
+    const optimisedJSON = { cs, ce, tc, h }
+
+    const cid = await client.uploadFile(
+      new Blob([JSON.stringify(optimisedJSON)], { type: 'application/json' }) as BlobLike
+    )
+    console.log(
+      `✅ Uploaded to IPFS Successfully for Cycles (${data.cycleStart} to ${data.cycleEnd}) @ https://${cid}.ipfs.w3s.link`
+    )
+    lastUploadTime = Date.now()
+    const storageUsed = await client.currentSpace()?.usage.get()
+    console.log(`${Number(storageUsed!.ok)} bytes used on Web3.Storage.`)
+    isPublisherActive = false
+  } catch (error) {
+    isPublisherActive = false
+    console.error(`❌ Error while Uploading Digest (w/ Hash: ${data.hash}) to IPFS: `)
```

Comment:
I included the hash only to keep the log line short and the hash makes it easier to search in a DB (if someone wants to quickly look it up since its just 1 value).
I'll add in start and end cycles too.